content/articles/generic-programming-in-c#/index.md

```diff
@@ -0,0 +1,182 @@
+
+### Introduction
+
+In any modern system, data handling is one of the major and vital goals. The way data is handled is what makes the total difference. Some systems go for generic one which includes lists, dictionaries, and hash.
+
+Others prefer non-generic which are arrays, array lists, and bit arrays. Non- generic programming represents an ordered collection of an object that can be indexed individually. As a matter of fact, generic programming is strongly typed while non- generic programming is not strongly typed, which means generic programming enforces strict rules on blending of values with different data types.
+
+But due to the great advancement of technology in the 21st century, many software systems require to store large groups of entities
+rather than individual entry. The concept of non-generic data handling can't fully solve this problem therefore in C#, generic collections are designed to solve this problem. Generic is a class that gives users the ability to define classes and methods with the placeholder.
+
+### Collections
+**Collections** according to C# are classes that hold together groups of objects. The `.NET` libraries provide useful system-defined classes, like collection namespace, which provide various collections and their functionalities. The use of non-generic collections is normally discouraged therefore my article fully focuses on generic collections.
+
+We have three common generic collections incorporated in the C# namespace.
+
+## 1. LIST
+
+A list is the most commonly used generic collection. Its because they are much more convenient when they are used instead of an array.
+A list is a very general data structure with each entry occupying a specific position known as an index. Lists are very flexible as they automatically resize every time data is added.
+
+Lists are very easy to manipulate. This is because many methods have been created and abstracted, meaning the bulky work is already done for the user. The user only has to call the method and manipulate the list.
+ 
+A list stores data in sequence, although the sequence oughtn't to have to be particular. It can also accept duplicate entries.
+Below is a pictorial explanation of a list. A generic list class is declared as shown:
+
+```
```

Comment:
> ```
> public class genericList<T>
> {
>     
>     void add(T input){}
> }
> ```

This code snippet is missing syntax highlighting add *csharp* after the upper 3 backticks (*`*) like so:

> \`\`\`csharp
> // your code goes here
> \`\`\`

so your code has syntax highlighting like so:

```csharp
using System;
using System.Collections.Generic;  
  
class MyDictinary {
    static public void Main () {
          
      
        //  Creating a dictionary using Dictionary <TKey,TValue> class
        Dictionary<int, string> My_dict1 = new Dictionary<int, string>(); 
            
          // Adding <key/value> pair in the Dictionary using Add() method
          My_dict1.Add(1123, "Welcome");
          My_dict1.Add(1124, "Hello");
          My_dict1.Add(1125, "World");
            
          foreach(KeyValuePair<int, string> ele1 in My_dict1)
          {
              Console.WriteLine("{0} and {1}", ele1.Key, ele1.Value);
          }
          Console.WriteLine();
            
          // Creating another dictionary
 
      Dictionary<string, string> My_dict2 = new Dictionary<string, string>(){
                                  {"a.1", "CSharp"},
                                  {"a.2", "Java"},
                                {"a.3", "Python"} }; 
           
          foreach(KeyValuePair<string, string> ele2 in My_dict2)
          {
              Console.WriteLine("{0} and {1}", ele2.Key, ele2.Value);
          }
    }
}
```